InnerEye/ML/Histopathology/models/deepmil.py

```diff
@@ -252,11 +254,15 @@ def _shared_step(self, batch: Dict, batch_idx: int, stage: str) -> Dict[ResultsK
 
         results = dict()
         for metric_object in self.get_metrics_dict(stage).values():
-            metric_object.update(predicted_probs, bag_labels)
+            if self.n_classes > 1:
+                metric_object.update(predicted_labels, bag_labels)
```

Comment:
shouldn't this be probs_per_class?

Reply:
no, probs_perclass is a vector. This remains the same

Reply:
I have now made change to accommodate binary and multi-class